pkg/apis/config/webhook/v1alpha1/types.go

```diff
@@ -25,11 +27,11 @@ type WebhookConfiguration struct {
 
 	// securePort is the port number to listen on for secure TLS connections from the kube-apiserver.
 	// Defaults to 6443.
-	SecurePort *int `json:"securePort,omitempty"`
+	SecurePort *int32 `json:"securePort,omitempty"`
 
 	// healthzPort is the port number to listen on (using plaintext HTTP) for healthz connections.
 	// Defaults to 6080.
-	HealthzPort *int `json:"healthzPort,omitempty"`
+	HealthzPort *int32 `json:"healthzPort,omitempty"`
```

Comment:
suggestion (non-blocking): Presumably in v1alpha2 we'd change this to be a regular int32, or a uint16. Is it worth documenting that the type will change in a future version in the comments?

Reply:
I think this has to be a pointer in the public type because if it is empty/ not set it should default to eg. 6080.
Also, we cannot use 0 as an empty value because that is the "pick any port" value.